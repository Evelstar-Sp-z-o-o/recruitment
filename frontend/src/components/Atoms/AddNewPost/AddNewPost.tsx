import { FC } from 'react';
import { useTranslation } from 'react-i18next';

import AddIcon from '@mui/icons-material/Add';
import IconButton from '@mui/material/IconButton';
import Tooltip from '@mui/material/Tooltip';
import { blue, grey } from '@mui/material/colors';

interface IAddNewPostProps {
  isFixed?: boolean;
}

const AddNewPost: FC<IAddNewPostProps> = ({ isFixed, onClick }) => {
  const StyledIconButton = {
    position: isFixed ? 'fixed' : 'static',
    bottom: '3rem',
    right: '1rem',
    height: '5rem',
    width: '5rem',
  };

  const StyledIcon = {
    color: grey['A100'],
    fontSize: '3rem',
    bgcolor: blue[700],
    borderRadius: '50%',
    p: 0,
  };
  const { t } = useTranslation();

  return (
    <Tooltip title={t('addPost')}>
      <IconButton aria-label={t('addPost')} role="button" sx={StyledIconButton} onClick={onClick}>
        <AddIcon sx={StyledIcon} />
      </IconButton>
    </Tooltip>
  );
};

export default AddNewPost;
