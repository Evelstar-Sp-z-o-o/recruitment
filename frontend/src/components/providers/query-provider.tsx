import { FC } from 'react';

import { QueryClientProvider, QueryClient } from '@tanstack/react-query';

const queryClient = new QueryClient();

interface Props {
  children: React.ReactNode;
}

const QueryProvider: FC<Props> = ({ children }) => {
  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

export default QueryProvider;
