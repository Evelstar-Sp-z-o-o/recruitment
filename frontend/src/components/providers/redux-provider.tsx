import store from "@/src/redux/store";
import { Provider } from "react-redux";

interface Props {
    children: React.ReactNode;
}

const ReduxProvider: React.FC<Props> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
};

export default ReduxProvider;
