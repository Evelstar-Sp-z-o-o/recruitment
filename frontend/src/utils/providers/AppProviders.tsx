import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import i18next from 'i18next';

import { store } from '@/src/store';

const AppProviders = ({ children }) => {
  return (
    <I18nextProvider i18n={i18next}>
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    </I18nextProvider>
  );
};

export default AppProviders;
