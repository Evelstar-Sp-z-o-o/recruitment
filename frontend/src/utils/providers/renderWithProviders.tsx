import { JSX, PropsWithChildren } from 'react';

import { render } from '@testing-library/react';

import AppProviders from './AppProviders.tsx';

export const renderWithProviders = (children: PropsWithChildren<JSX.Element>) => {
  return render(<AppProviders>{children}</AppProviders>);
};
