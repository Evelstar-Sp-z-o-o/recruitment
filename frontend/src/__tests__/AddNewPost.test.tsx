import React from 'react';

import AddNewPost from '@/src/components/Atoms/AddNewPost/AddNewPost';
import { renderWithProviders } from '@/src/utils/providers/renderWithProviders';
import { blue, grey } from '@mui/material/colors';
import '@testing-library/jest-dom';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

vi.mock('react-i18next', async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    I18nextProvider: ({ children }) => <div>{children}</div>,
  };
});

describe('AddNewPost', () => {
  it('renders AddNewPost component', () => {
    renderWithProviders(<AddNewPost />);

    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
  });

  it('renders with tooltip', () => {
    renderWithProviders(<AddNewPost />);

    const button = screen.getByRole('button');
    expect(button).toHaveAttribute('aria-label', 'addPost');
  });

  it('calls onClick when button is clicked', async () => {
    const handleClick = vi.fn();
    renderWithProviders(<AddNewPost onClick={handleClick} />);

    const button = screen.getByRole('button');
    userEvent.click(button);

    await waitFor(() => expect(handleClick).toHaveBeenCalledTimes(1));
  });

  it('applies fixed positioning when isFixed is true', () => {
    renderWithProviders(<AddNewPost isFixed />);

    const button = screen.getByRole('button');
    expect(button).toHaveStyle({
      position: 'fixed',
      bottom: '3rem',
      right: '1rem',
      height: '5rem',
      width: '5rem',
    });
  });

  it('applies static positioning when isFixed is false', () => {
    renderWithProviders(<AddNewPost isFixed={false} />);

    const button = screen.getByRole('button');
    expect(button).toHaveStyle({
      position: 'static',
      bottom: '3rem',
      right: '1rem',
      height: '5rem',
      width: '5rem',
    });
  });

  it('renders AddIcon with correct styles', () => {
    renderWithProviders(<AddNewPost />);

    const icon = screen.getByRole('button').firstChild;
    expect(icon).toHaveStyle({
      color: grey['A100'],
      fontSize: '3rem',
      backgroundColor: blue[700],
      borderRadius: '50%',
      padding: 0,
    });
  });

  it('should match snapshot', () => {
    const snapshot = renderWithProviders(<AddNewPost />);
    expect(snapshot).toMatchSnapshot();
  });
});
