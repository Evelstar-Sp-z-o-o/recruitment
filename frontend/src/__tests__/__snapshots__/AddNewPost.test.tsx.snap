// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddNewPost > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          aria-label="addPost"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wpnq05-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          role="button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16960id-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        aria-label="addPost"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wpnq05-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        role="button"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-16960id-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
